src/core.js

```diff
@@ -1288,18 +1289,33 @@ define([
                 }
             }
         });
-        this.selectSomethingOrHideProperties(true);
+
+        if (qId) {
+            var mug = _this.getMugByPath(qId);
+            if (mug) {
+                _this.selectSomethingOrHideProperties(true, mug.ufid);
+            } else {
+                _this.selectSomethingOrHideProperties(true);
+            }
+        } else {
+            _this.selectSomethingOrHideProperties(true);
+        }
+
     };
 
-    fn.selectSomethingOrHideProperties = function (forceDeselect) {
+    fn.selectSomethingOrHideProperties = function (forceDeselect, ufid) {
         if (forceDeselect) {
             this.jstree('deselect_all');
         }
         // ensure something is selected if possible
         if (!this.jstree('get_selected').length) {
             // if there's any nodes in the tree, just select the first
-            var all_nodes = this.data.core.$tree.find("li");
-            if (all_nodes.length > 0) {
+            var all_nodes = this.data.core.$tree.find("li"),
+                selected = all_nodes.filter('[id= ' + ufid + ']');
```

Comment:
Does it matter if `ufid` is null or undefined? Seems like it might be better to skip the filter if `ufid` evaluates to false.